src/hooks/useCalculate.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useRef, useState } from 'react';
+import { operations } from '../constants';
+import { calculate } from '../utils';
+
+const useCalculate = () => {
+  const [total, setTotal] = useState<number | null>(null);
+
+  const operatorRef = useRef<(next: number) => number>();
+  const inputCountRef = useRef<number>(0);
+
+  const handleOperationClick = (operation: keyof typeof operations) => {
+    if (total === null) return;
+
+    if (inputCountRef.current === 0) {
+      operatorRef.current = calculate[operation](total);
+      return;
+    }
+
+    const nextTotalValue = operatorRef.current
+      ? Math.floor(operatorRef.current(total))
+      : total;
+
+    setTotal(nextTotalValue);
+    operatorRef.current = calculate[operation](nextTotalValue);
+
+    inputCountRef.current = 0;
+  };
+
+  const handleDigitClick = useCallback((digit: number) => {
+    const count = inputCountRef.current;
+
+    setTotal((prevTotal) => {
+      if (!prevTotal) return digit;
```

Comment:
```suggestion
      if (prevTotal === null) return digit;
```
이번 케이스에서는 0이 들어왔을 때 바로 digit를 반환하기에 괜찮겠지만 falsy한 값을 다룰때는 null 체크를 더 좁게 체크해주는게 좋아요.
의도치 않은 동작을 할 여지가 있기에 [lint](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md)의 도움을 받는것도 추천드려요~

Reply:
[9c3340a](https://github.com/next-step/react-calculator/pull/4/commits/9c3340a2499fda82eaaf66ae42ddbab168dc26d5) 여기서 린트 설정해주었고 
[17c7de6](https://github.com/next-step/react-calculator/pull/4/commits/17c7de67b4828a320e2c13a04c0f4bfc5fa2cbc5) 여기서 린트관련해서 수정해주었어요